doc: update HomeKit known issues (#11846)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

.. rst-class:: v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

NCSDK-13947: Net core downgrade prevention does not work on nRF5340

>   Downgrade prevention mechanism does not work for Net core when update via SMP protocol (Nordic DFU) is used.

Reviewer: ```suggestion
  When updating firmware via SMP protocol (Nordic DFU), the downgrade prevention mechanism does not work for the network core.
```
